source/unified-test-format/unified-test-format.rst

```diff
@@ -1397,6 +1397,18 @@ The structure of each object is as follows:
   that the messages produced are an exact match, i.e. that the expected and actual
   message counts are the same and that there are no extra messages emitted by the
   client during the test run.
+  Note: ``ignoreMessages`` and ``ignoreExtraMessages`` may exclude log messages from this evaluation.
+
+- ``ignoreMessages``: Optional array of `expectedLogMessage`_ objects. Unordered set of
+  messages, which MUST be ignored on the corresponding client while executing `operations`_.
+  The test runner MUST exclude all log messages from observed messages that match any of the messages
+  in ``ignoreMessages`` array before ``messages`` evaluation.
+  Matching rules used to match messages in ``ignoreMessages`` are identical to match rules used for ``messages`` matching.
```

Comment:
Should we relax the "ignoreMessages" conditions?

>  Matching rules used to match messages in ``ignoreMessages`` are identical to match rules used for ``messages`` matching.

Something like this:

> Matching rules used to match messages in ``ignoreMessages`` are upto level, component, and data.message.

For example, in the logging-replicatset.yml unified spec test we have the following message in the ignore list:

```yml
- level: debug
  component: topology
  data:
    message: "Server heartbeat started"
```
However, according to the section ["Server Heartbeat Started" Log Message](https://github.com/mongodb/specifications/pull/1419/files?short_path=e2f3a7c&unchanged=expanded#id29) the unstructured log is this:

> Heartbeat started for {{serverHost}}:{{serverPort}} on connection with driver-generated ID {{driverConnectionId}} and server-generated ID {{serverConnectionId}} in topology with ID {{topologyId}}. Awaited: {{awaited}}

The above implies that serverHost, serverPort, serverConnectionId, topologyId and awaited fields should all be present in the "data" structure. Thus I would expect the structured analogue to always looks something like this:

```json
{"level": "debug", "component": "topology", "data": {"message": "Server heartbeat started" , "serverPort": "x", "serverHost": "y", "topologyId": "z", "awaited": true}}
```
And so the above yaml will never actually indicate ignoring anything (i.e. the data would always be a subset of the actual message's data).

Reply:
I think we are good here.
Matching the data field is done according to [root document evaluation rules](https://github.com/mongodb/specifications/blob/master/source/unified-test-format/unified-test-format.rst). Which permits extra fields. [expectedLogMessage](https://github.com/mongodb/specifications/blob/master/source/unified-test-format/unified-test-format.rst#id40)

Also we have to permit extra fields as drivers may add their own parameters as needed.